qa/sc_ft_limit_fork.py

```diff
@@ -0,0 +1,126 @@
+
+    def setup_nodes(self):
+        num_nodes = 1
+        # Set MC scproofqueuesize to 0 to avoid BatchVerifier processing delays
+        return start_nodes(num_nodes, self.options.tmpdir, extra_args=[['-debug=sc', '-debug=ws', '-logtimemicros=1',
+                                                                        '-scproofqueuesize=0']] * num_nodes)
+
+    def sc_setup_chain(self):
+        # After bug spotted in 0.3.4 we test certificate generation with max keys number > 8
+        cert_max_keys = 10
+        cert_sig_threshold = 6
+
+        mc_node = self.nodes[0]
+        sc_node_configuration = SCNodeConfiguration(
+            MCConnectionInfo(address="ws://{0}:{1}".format(mc_node.hostname, websocket_port_by_mc_node_index(0))),
+            submitter_private_keys_indexes=list(range(cert_max_keys))  # SC node owns all schnorr private keys.
+        )
+
+        network = SCNetworkConfiguration(SCCreationInfo(mc_node, 100, self.sc_withdrawal_epoch_length,
+                                                        cert_max_keys=cert_max_keys,
+                                                        cert_sig_threshold=cert_sig_threshold), sc_node_configuration)
+        self.sc_nodes_bootstrap_info = bootstrap_sidechain_nodes(self.options, network, 720 * 120 * 10)
+
+    def sc_setup_nodes(self):
+        return start_sc_nodes(1, self.options.tmpdir)
+
+    def run_test(self):
+        time.sleep(0.1)
+        self.sync_all()
+        mc_node = self.nodes[0]
+        sc_node = self.sc_nodes[0]
+
+        self.do_ft(0.00000050)
+        self.generate_certificate()
+
+        # check we are still in pre-fork epoch and cert has 0 ftScFee
+        forging_info = http_block_forging_info(sc_node)
+        assert_equal(forging_info["bestEpochNumber"], 2)
+        we0_cert_hash = mc_node.getrawmempool()[0]
+        we0_cert = mc_node.getrawtransaction(we0_cert_hash, 1)
+        assert_equal(we0_cert["cert"]["ftScFee"], 0)
+
+        # switch to the next consensus epoch, generate new cert, check that ftScFee is 54 now
+        generate_next_block(sc_node, "first node", force_switch_to_next_epoch=True)
+        forging_info = http_block_forging_info(sc_node)
+        assert_equal(forging_info["bestEpochNumber"], 3)
+        self.generate_certificate()
+        we0_cert_hash = mc_node.getrawmempool()[0]
+        print("Withdrawal epoch 0 certificate hash = " + we0_cert_hash)
+        we0_cert = mc_node.getrawtransaction(we0_cert_hash, 1)
+        assert_equal(we0_cert["cert"]["ftScFee"], Decimal(54) / Decimal(100000000))
+
+        # skip a couple consensus epochs, try to do another FT with 50 satoshi, assert it fails
+        generate_next_block(sc_node, "first node", force_switch_to_next_epoch=True)
+        generate_next_block(sc_node, "first node", force_switch_to_next_epoch=True)
+        generate_next_block(sc_node, "first node", force_switch_to_next_epoch=True)
+
+        assert_raises(AssertionError, self.do_ft, 0.00000050)
```

Comment:
Please also check that FT with >=54 satoshi will be added to mempool in this point.
Also it's better to check that all FTs successfully arrived to SC, please check cons balance with check_wallet_coins_balance or check_box_balance.